storage/storage_reference.go

```diff
@@ -30,6 +30,16 @@ func newReference(transport storageTransport, named reference.Named, id string)
 	if named == nil && id == "" {
 		return nil, ErrInvalidReference
 	}
+	if named != nil {
```

Comment:
Nit: Why not use the && here?  Used it above.

if named != nil && reference.IsNameOnly(named) {
 			return nil, errors.Wrapf(ErrInvalidReference, "reference %s has neither a tag nor a digest", named.String())
}

Reply:
Thanks, fixed.